### PR TITLE
Initial cppfs commit

### PR DESCRIPTION
**Pull request by polyntsov**

### CODE REVIEW

File: cpp-fs/cmake/BuildOptions.cmake [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+)
+
+set(ASAN_LINK_OPTIONS
+    -fsanitize=address
+)
+
+add_compile_options("$<$<CONFIG:ASAN>:${ASAN_BUILD_OPTIONS}>")
+add_link_options("$<$<CONFIG:ASAN>:${ASAN_LINK_OPTIONS}>")
+
+add_compile_options(-fmodules-ts -MMD)
```
Comment (polyntsov) [inline cpp-fs/cmake/BuildOptions.cmake]: oops

File: cpp-fs/cmake/BuildOptions.cmake [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    -Werror
+    -Wpedantic
+)
+
+add_compile_options("$<$<CONFIG:Release>:${RELEASE_BUILD_OPTIONS}>")
+
+set(ASAN_BUILD_OPTIONS
+    -O0
+    -Wall
+    -Wall
```
Comment (vs9h) [inline cpp-fs/cmake/BuildOptions.cmake]: duplicate

File: cpp-fs/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.28)
```
Comment (vs9h) [inline cpp-fs/CMakeLists.txt]: go downgrade to 3.27
on the latest Ubuntu version 3.27.4

File: cpp-fs/.clang-format [hunk 4]
```diff
@@ -0,0 +1,11 @@
+BasedOnStyle: Google
+IncludeCategories:
+  - Regex: '<boost.+>'
+    Priority: 2
+  - Regex: '<.+>'
+    Priority: 1
+  - Regex: '.+'
+    Priority: 3
+QualifierAlignment: Custom
+QualifierOrder: ['inline', 'static', 'type', 'const', 'volatile' ]
```
Comment (vs9h) [inline cpp-fs/.clang-format]:
```suggestion
QualifierOrder: ['inline', 'static', 'type', 'const', 'volatile']
```

**Review by vs9h — APPROVED** [review]

**Pull request merged**